A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1092,25 +1106,44 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
     }
     checkUpdateFlightPhase() {
         const airSpeed = SimVar.GetSimVarValue("AIRSPEED TRUE", "knots");
+        const groundSpeed = SimVar.GetSimVarValue("GPS GROUND SPEED", "knots");
         const flapsHandlePercent = Simplane.getFlapsHandlePercent();
         const leftThrottleDetent = Simplane.getEngineThrottleMode(0);
         const rightThrottleDetent = Simplane.getEngineThrottleMode(1);
         const highestThrottleDetent = (leftThrottleDetent >= rightThrottleDetent) ? leftThrottleDetent : rightThrottleDetent;
 
         //End preflight when takeoff power is applied and engines are running
-        if (this.currentFlightPhase <= 2) {
-            if ((highestThrottleDetent == ThrottleMode.TOGA || highestThrottleDetent == ThrottleMode.FLEX_MCT) && SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent") > 15 && SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent") > 15) {
+        if (this.currentFlightPhase < FlightPhase.FLIGHT_PHASE_TAKEOFF) {
+            if (
+                (highestThrottleDetent === ThrottleMode.TOGA || highestThrottleDetent === ThrottleMode.FLEX_MCT) &&
+                (
+                    SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent") > 85 &&
+                    SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent") > 85
+                ) || groundSpeed > 90
+            ) {
                 SimVar.SetSimVarValue("L:A32NX_Preflight_Complete", "Bool", 1);
+                this.currentFlightPhase = FlightPhase.FLIGHT_PHASE_TAKEOFF;
             }
         }
 
         //Reset to preflight in case of RTO
-        if (this.currentFlightPhase <= 2 && SimVar.GetSimVarValue("L:A32NX_Preflight_Complete", "Bool") == 1) {
-            if (!(highestThrottleDetent == ThrottleMode.TOGA || highestThrottleDetent == ThrottleMode.FLEX_MCT) && SimVar.GetSimVarValue("RADIO HEIGHT", "Feet") < 100) {
+        if (this.currentFlightPhase === FlightPhase.FLIGHT_PHASE_TAKEOFF) {
+            if (
+                SimVar.GetSimVarValue("RADIO HEIGHT", "Feet") < 100 &&
```

Comment:
I wasn't able to find a reference for this line. I'll just leave it as is for now, and invert the preflight => takeoff conditions below so we don't end up in a situation where the above preflight => takeoff condition fights with the takeoff => preflight conditions here.